src/tests.cpp

```diff
@@ -1424,3 +1438,52 @@ TEST_CASE( "BSMHandler JSON Error Checking", "[ppm][filtering][error]" ) {
         CHECK_FALSE( handler.process( test_case ) );
     }
 }
+
+TEST_CASE( "RedactionPropertiesManager", "[ppm][redaction][properties]") {
+    RedactionPropertiesManager redactionPropertiesManager;
+    CHECK ( redactionPropertiesManager.getNumFields() > 0);
+}
+
+TEST_CASE( "BSMHandler JSON PartII Redaction Only", "[ppm][filtering][partIIonly]" ) {
+    RedactionPropertiesManager rpm;
+
+    std::unordered_map<std::string,std::string> pconf;
+
+    REQUIRE( buildBaseConfiguration( pconf ) ); 
+    BSMHandler handler{ buildTestQuadTree(), pconf };
+
+    handler.deactivate<BSMHandler::kVelocityFilterFlag>();
+    handler.deactivate<BSMHandler::kGeofenceFilterFlag>();
+    handler.deactivate<BSMHandler::kIdRedactFlag>();
+
+    // activate kPartIIRedactFlag and make sure the other flags are deactivated
+    REQUIRE( handler.is_active<BSMHandler::kPartIIRedactFlag>() );
+    REQUIRE_FALSE( handler.is_active<BSMHandler::kVelocityFilterFlag>() );
+    REQUIRE_FALSE( handler.is_active<BSMHandler::kGeofenceFilterFlag>() );
+    REQUIRE_FALSE( handler.is_active<BSMHandler::kIdRedactFlag>() );
+
+    std::vector<std::string> json_test_cases;
+    REQUIRE ( loadTestCases( "unit-test-data/test-case.partII.json", json_test_cases ) );
+
+    for ( auto& test_case : json_test_cases ) {
+
+        // process test case to build BSM (redaction will occur here)
+        CHECK( handler.process( test_case ) );
+
+        // make sure that it was successful
+        CHECK( handler.get_result_string() == "success" );
+
+        // verify that there are no sensitive members in the partII field left
+        std::string partIIString = handler.get_bsm().get_partII();
+        rapidjson::Document partII;
+        partII.Parse(partIIString.c_str());
```

Comment:
This needs to be verified to have correctly parsed the partII string. If it is not getting parsed correctly, which I suspect is the case, then it's likely that we need to alter the partIIString before we parse it.

Reply:
To clarify, redaction functionality has been verified to be working via debug messages, but I don't think the above code will cause the test to fail (as it should) when there are still fields that should have been redacted remaining in the partII section of the BSM message.

Reply:
This test will now fail upon failure to parse and upon finding sensitive fields in the partII section of the BSM message after redaction. The test also requires that the partIIString not be empty.